### ISSUE TITLE
[Outdated] citra-canary-git

### ISSUE THREAD

**Issue by niobium93** [chaotic-aur/packages#2505]
### Outdated Package

[citra-canary-git](https://aur.archlinux.org/packages/citra-canary-git)

### Latest chaotic version

'r9536.631c3f66a-1'

### Latest upstream version

'r9577.399ed10f3-1'

### Logs for the latest build, if available

[citra-canary-git](https://builds.garudalinux.org/repos/chaotic-aur/logs/citra-canary-git.log)

### More information

```
==> ERROR: /home/main-builder/pkgsrc/soundtouch is not a clone of https://codeberg.org/soundtouch/soundtouch.git
```

**Comment by Technetium1** [comment]
Cache was bad, thank you for reporting this! Still failing for other reasons now though.
```
CMake Error at src/video_core/host_shaders/CMakeLists.txt:32 (message):
  Required program `glslangValidator` not found.
-- Configuring incomplete, errors occurred!
```

**Comment by niobium93** [comment]
The dependency `glslang` has been added. It also looks like chaotic already built it successfully as the package updated on my end, though the log still seems outdated?

**Comment by fwookerton** [comment]
Latest chaotic version: 'r9621.2c494c74b-1'
Latest upstream version: '2545.r0.g1872ada-1'

Cache failed again...
https://builds.garudalinux.org/repos/chaotic-aur/logs/citra-canary-git.log

**Comment by fwookerton** [comment]
@Technetium1 @xiota

**Comment by xiota** [comment]
I do prefer comments on old, relevant issues, rather than opening new issues.  It reduces duplicates and notifies everyone who's interested in the issue.

I've rebuilt `citra-canary-git` (2568.r0.gbfcce2f-1) and `citra-git` (r9707.22c4eb86d-1).  `citra-qt-git` is part of `citra-git`.

**Comment by fwookerton** [comment]
this package consistently errors when building lol

**Comment by xiota** [comment]
Builds okay on dragon.

**Comment by dr460nf1r3** [comment]
Yeah, the reason here being that the actual builder (`ufscar-hpc`) has a another git repo in cache than the PKGBUILD expects. `chaotic-dragon` does not have the faulty cache so it succeeds 😸
Definitely calls for removing that particular cache of the builder to permanantly fix it.

**Comment by xiota** [comment]
`citra-canary-git` built earlier today segfaults.  Guessing an upstream issue.  Until it's fixed, use `citra-qt-git`, which was not rebuilt.

**Comment by Technetium1** [comment]
@xiota There's not really a way to ensure it's working or not at build time, just if it's passing the tests. Ideally we always have last working `-git` packages, because build must complete to upload, and the build should be always doing `check()`.

If a package breaks, there is [was?] a way to roll back to last, but I have never been the one to do it, and I'm not sure if rolling back is added to the toolbox just yet; @dr460nf1r3 can hopefully give details?

**Comment by Technetium1** [comment]
It definitely seems like a great thing to add to the toolbox at some point!

**Comment by dr460nf1r3** [comment]
@xiota I found something interesting while running the ufscar routine:
~~~
Downloading: citra-canary-git                               Error: gzip: invalid header
~~~

**Comment by Technetium1** [comment]
Weird, [this line](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=citra-canary-git#n106) seems to include it in the loop that works for others.

**Comment by dr460nf1r3** [comment]
It works as long as the git repository doesn't change when it's cached. Guess I'll put those to build on `garuda-cluster` instead since clearing sources is easier there.

**Comment by xiota** [comment]
Build issues appear to have been resolved.  `citra-canary-git` last built 2023-09-14.
